server/gameserver.js

```diff
@@ -102,7 +107,34 @@ async function SharedTryAddServer( request )
 
 	if ( typeof request.query.authPort == "string" )
 		request.query.authPort = parseInt( request.query.authPort )
-
+	// Verify string is ASCII only
+	if (
+		request.query.name.match( ASCIIRegex ) ||
+		request.query.description.match( ASCIIRegex ) ||
+		request.query.map.match( ASCIIRegex ) ||
+		request.query.playlist.match( ASCIIRegex )
+	)
+	{
+		return { success: false, error: INVALID_STRING_DATA }
+	}
+	let counter = 0 // Usewd to track servers per ip, could maybe be optimized into a Hashmap(ip, counter)
+	let servers = GetGameServers()
+	for ( let key in servers )
+	{
+		let server = servers[ key ]
+		if ( server.ip == request.ip )
+		{
+			if ( server.port == request.query.port ) // Block if server with that ip and port already exists
+			{
+				return { success: false, error: DUPLICATE_SERVER }
+			}
```

Comment:
Could we make it so it overwrites the existing server entry? If a server crashes and auto restarts, this will prevent it from updating.

Reply:
a server crashing should in theory make a ghost server but i can't recall what the timeout for that is. but if a server restarted too fast it would cause this error